stan/math/prim/meta/holder.hpp

```diff
@@ -0,0 +1,274 @@
+#ifndef STAN_MATH_PRIM_FUN_HOLDER_HPP
+#define STAN_MATH_PRIM_FUN_HOLDER_HPP
+
+#include <stan/math/prim/meta/is_eigen.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+namespace stan {
+namespace math {
+
+template <class ArgType, typename... Ptrs>
+class Holder;
+
+}  // namespace math
+}  // namespace stan
+
+namespace Eigen {
+namespace internal {
+
+template <class ArgType, typename... Ptrs>
+struct traits<stan::math::Holder<ArgType, Ptrs...>> {
+  typedef typename ArgType::StorageKind StorageKind;
+  typedef typename traits<ArgType>::XprKind XprKind;
+  typedef typename ArgType::StorageIndex StorageIndex;
+  typedef typename ArgType::Scalar Scalar;
+  enum {
+    Flags = (ArgType::Flags
+             & (RowMajorBit | LvalueBit | LinearAccessBit | DirectAccessBit
```

Comment:
Isn't ```RowMajorBit``` redundant given that you've also specified ```NoPreferredStorageOrderBit```?

Reply:
They are two separate bits that are possible in flags. `NoPreferredStorageOrderBit`'s doc says it means transpositions are never needed when working with this expression or something like this.